src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/InstantiateArgumentExceptionsCorrectlyTests.cs

```diff
@@ -106,49 +106,148 @@ End Sub
         }
 
         [Fact]
-        public async Task ArgumentException_ParameterNameAsMessage_Warns()
+        public async Task ArgumentException_ParameterNameAsMessage_WarnsAndCodeFixesWithNameOf()
         {
-            await VerifyCS.VerifyAnalyzerAsync(@"
+            await VerifyCS.VerifyCodeFixAsync(@"
                 public class Class
                 {
                     public void Test(string first)
                     {
                         throw new System.ArgumentException(""first"");
                     }
                 }",
-                GetCSharpIncorrectMessageExpectedResult(6, 31, "Test", "first", "message", "ArgumentException"));
+                GetCSharpIncorrectMessageExpectedResult(6, 31, "Test", "first", "message", "ArgumentException"), @"
+                public class Class
+                {
+                    public void Test(string first)
+                    {
+                        throw new System.ArgumentException(null, nameof(first));
+                    }
+                }");
 
-            await VerifyVB.VerifyAnalyzerAsync(@"
-                Public Class [MyClass]
-                    Public Sub Test(first As String)
-                        Throw New System.ArgumentException(""first"")
-                    End Sub
-                End Class",
-                GetBasicIncorrectMessageExpectedResult(4, 31, "Test", "first", "message", "ArgumentException"));
+            await VerifyVB.VerifyCodeFixAsync(@"
+Public Class [MyClass]
+    Public Sub Test(first As String)
+        Throw New System.ArgumentException(""first"")
+    End Sub
+End Class",
+                GetBasicIncorrectMessageExpectedResult(4, 15, "Test", "first", "message", "ArgumentException"), @"
+Public Class [MyClass]
```

Comment:
Somehow indented sources not matching for VB tests, not sure why, will try to fix later

Reply:
it seems not really depending from my code/fixer, Document editor might have issue with replacing VB node, @mavasani any suggestions?